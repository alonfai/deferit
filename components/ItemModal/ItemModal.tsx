import React from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';
import RNModal from 'react-native-modal';
import WebModal from 'react-modal';
import { helpers } from '../../utils';

export type Props = {
  /**
   * Show the modal dialgo
   */
  show: boolean;
  /**
   * SetState event dispatch hook to show/hide the modal
   */
  setShow: React.Dispatch<React.SetStateAction<boolean>>;
};

const ItemModal: React.FC<Props> = ({ children, show, setShow }) => {
  const device = helpers.getDevice();

  // render a different modal component based on what device platform we using
  return device === 'Mobile' ? (
    <RNModal isVisible={show}>
      <TouchableOpacity
        style={styles.modalContainer}
        onPress={() => {
          setShow(false);
        }}
      >
        <TouchableOpacity activeOpacity={1}>{children}</TouchableOpacity>
      </TouchableOpacity>
    </RNModal>
  ) : (
    <WebModal
      isOpen={show}
      onRequestClose={() => {
        setShow(false);
      }}
      style={{
        content: {
          top: '50%',
          left: '50%',
          right: 'auto',
          bottom: 'auto',
          marginRight: '-50%',
          transform: 'translate(-50%, -50%)',
          maxWidth: '800px',
          maxHeight: '800px',
          borderRadius: 10,
          borderWidth: 0.5,
        },
      }}
    >
      {children}
    </WebModal>
  );
};
const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default ItemModal;
