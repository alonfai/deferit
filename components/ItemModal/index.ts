export { default } from './ItemModal';
export type { Props } from './ItemModal';
