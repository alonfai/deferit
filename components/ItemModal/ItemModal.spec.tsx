/// <reference types="@testing-library/jest-native" />
import React from 'react';
import { render } from '@testing-library/react-native';
import ItemModal from './ItemModal';

const mockGetDevice = jest.fn();
const mockSetShow = jest.fn();
jest.mock('../../utils', () => {
  return {
    helpers: {
      getDevice: () => mockGetDevice(),
    },
  };
});

describe('<ItemModal />', () => {
  test('Render successfully', async () => {
    mockGetDevice.mockReturnValue('Web');
    const { toJSON } = render(
      <ItemModal show setShow={mockSetShow}>
        Hello World
      </ItemModal>
    );
    expect(mockGetDevice).toBeCalled();
    expect(toJSON()).toMatchSnapshot();
  });
});
