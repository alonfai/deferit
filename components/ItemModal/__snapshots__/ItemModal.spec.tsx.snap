// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemModal /> Render successfully 1`] = `null`;
